server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/SetMailboxesUpdateProcessor.java

```diff
@@ -156,9 +164,19 @@ private void handleUpdate(MailboxId mailboxId, MailboxUpdateRequest updateReques
         }
    }
 
-    private void checkRole(Optional<Role> role) throws SystemMailboxNotUpdatableException {
-        if (role.map(Role::isSystemRole).orElse(false)) {
-            throw new SystemMailboxNotUpdatableException();
+    private void assertNotSharedOutboxOrDraftMailbox(Mailbox mailbox, MailboxUpdateRequest updateRequest) {
+        Preconditions.checkArgument(!updateRequest.getSharedWith().isPresent() || !mailbox.hasRole(Role.OUTBOX), "Cannot share 'Outbox'");
```

Comment:
these conditions are still not very clear (and I think I prefered with &&), so can you extract these conditions to something like
```
boolean isSharedWithRole(Role role, ...) {
    return updateRequest.getSharedWith().isPresent() && mailbox.hasRole(role);
}

Reply:
I did it but it's easier with code inlined IMO.